drivers/pca9633/pca9633.c

```diff
@@ -0,0 +1,266 @@
+{
+    assert(dev);
+    assert(params);
+
+    dev->params = *params;
+
+    i2c_init(dev->params.i2c_dev);
+
+    int rc = _write_reg(dev, PCA9633_REG_MODE1, 0x0);
+    _write_reg(dev, PCA9633_REG_MODE2, 0x0);
+
+    if (rc != PCA9633_OK) {
+        return rc;
+    }
+
+    pca9633_set_ldr_state_all(dev, PCA9633_LDR_STATE_IND_GRP);
+    pca9633_set_group_control_mode(dev, PCA9633_GROUP_CONTROL_MODE_DIMMING);
+    pca9633_set_rgb(dev, 255, 255, 255);
+
+    return PCA9633_OK;
+}
+
+void pca9633_wakeup(pca9633_t* dev)
+{
+    uint8_t prev_reg;
+    _read_reg(dev, PCA9633_REG_MODE1, &prev_reg);
+    uint8_t newReg = prev_reg & ~(1 << PCA9633_BIT_SLEEP);
+
+    _write_reg(dev, PCA9633_REG_MODE1, newReg);
+}
+
+void pca9633_sleep(pca9633_t* dev)
+{
+    uint8_t prev_reg;
+    _read_reg(dev, PCA9633_REG_MODE1, &prev_reg);
+    uint8_t newReg = prev_reg | (1 << PCA9633_BIT_SLEEP);
+
+    _write_reg(dev, PCA9633_REG_MODE1, newReg);
+}
+
+void pca9633_turn_on(pca9633_t* dev)
+{
+    _write_reg(dev, PCA9633_REG_LEDOUT, dev->stored_reg_ledout);
+}
+
+void pca9633_turn_off(pca9633_t* dev)
+{
+    _read_reg(dev, PCA9633_REG_LEDOUT, &dev->stored_reg_ledout);
+    _write_reg(dev, PCA9633_REG_LEDOUT, PCA9633_LDR_STATE_OFF);
+}
+
+void pca9633_set_pwm(pca9633_t* dev, uint8_t reg_pwm, uint8_t pwm)
+{
+    _write_reg(dev, reg_pwm, pwm);
+}
+
+void pca9633_set_grp_pwm(pca9633_t* dev, uint8_t pwm)
+{
+    _write_reg(dev, PCA9633_REG_GRPPWM, pwm);
+}
+
+void pca9633_set_blinking(pca9633_t* dev, uint8_t blink_period,
+        float on_off_ratio)
+{
+    int16_t ratio = on_off_ratio * 256;
+
+    if (ratio < 0) {
+        ratio = 0;
+    }
+    else if (ratio > 255) {
+        ratio = 255;
+    }
+
+    _write_reg(dev, PCA9633_REG_GRPFREQ, blink_period);
+    _write_reg(dev, PCA9633_REG_GRPPWM, (uint8_t) ratio);
+}
+
+void pca9633_set_rgb(pca9633_t* dev, uint8_t r, uint8_t g, uint8_t b)
+{
+    pca9633_set_pwm(dev, dev->params.reg_pwm_red, r);
+    pca9633_set_pwm(dev, dev->params.reg_pwm_green, g);
+    pca9633_set_pwm(dev, dev->params.reg_pwm_blue, b);
+}
+
+void pca9633_set_rgba(pca9633_t* dev, uint8_t r, uint8_t g, uint8_t b, uint8_t a)
+{
+    pca9633_set_rgb(dev, r, g, b);
+
+    if (dev->params.has_amber_channel) {
+        pca9633_set_pwm(dev, dev->params.reg_pwm_amber, a);
+    }
+}
+
+void pca9633_set_ldr_state(pca9633_t* dev, uint8_t state, uint8_t ldr_bit)
+{
+    uint8_t prev_reg;
+    _read_reg(dev, PCA9633_REG_LEDOUT, &prev_reg);
+    uint8_t new_reg;
+
+    // first clear both bits of ldr
+    new_reg = prev_reg & ~(0b11 << ldr_bit);
+
+    // second set new state to specified ldr
+    new_reg |= (state << ldr_bit);
+
+    _write_reg(dev, PCA9633_REG_LEDOUT, new_reg);
+}
+
+void pca9633_set_ldr_state_all(pca9633_t* dev, uint8_t state)
+{
+    uint8_t new_reg = ( (state << PCA9633_BIT_LDR3)
+                       | (state << PCA9633_BIT_LDR2)
+                       | (state << PCA9633_BIT_LDR1)
+                       | (state << PCA9633_BIT_LDR0) );
```

Comment:
```suggestion
    uint8_t new_reg = (state << PCA9633_BIT_LDR3)
                    | (state << PCA9633_BIT_LDR2)
                    | (state << PCA9633_BIT_LDR1)
                    | (state << PCA9633_BIT_LDR0);
```

excessive parenthesis